### ISSUE TITLE
Graphql mutation returning null when giving a siteID

### ISSUE THREAD

**Issue by edwwaarrdd** [craftcms/cms#7468]
### Description

When adding en entry through a mutation I get null as a return when I'm using a siteId in the query

### Steps to reproduce

1. Try a mutation with siteID 1 and it will return correct output
Input:

```
mutation saveProduct{
    save_products_products_Entry(
      authorId: 1
      siteId: 1
      productName: "abc"
    ){
     id
     productName
    }
  }
```

  Output:

```
{
  "data": {
    "save_products_products_Entry": {
      "id": "46853",
      "productName": "abc"
    }
  }
}
```

2. Try with siteID 2:
Input:

```
mutation saveProduct{
    save_products_products_Entry(
      authorId: 1
      siteId: 2
      productName: "abc"
    ){
     id
     productName
    }
  }
```

  Output:

```
  {
  "data": {
    "save_products_products_Entry": null
  }
}
```

And expected output would also be 

```
{
  "data": {
    "save_products_products_Entry": {
      "id": "46853",
      "productName": "abc"
    }
  }
}
```

### Additional info

- Craft version: Craft Pro 3.5.17.1
- PHP version: 7.4.13
- Database driver & version: PostgreSQL 13.1
- Plugins & versions:

**Comment by edwwaarrdd** [comment]
This user on stack overflow has the same issue:
https://craftcms.stackexchange.com/questions/36611/how-can-i-have-created-data-id-returned-for-a-graphql-mutation

**Comment by andris-sevcenko** [comment]
@edwwaarrdd is the entry saved in the second case? If not, perhaps the section is not enabled for the site in question?

If none of that provides any answers - is there anything in the error logs? If not, is it an option to update to Craft 3.6.2? One of the changes there is improved error logging for GraphQL operations, which should help track this down.

**Comment by edwwaarrdd** [comment]
@andris-sevcenko 
Yes the entry is saved and I can fetch it with a query request to the graphql endpoint.
I will try to update the instance locally to 3.6.2 today and see if I can find anything in the logs.

**Comment by andris-sevcenko** [comment]
@edwwaarrdd looking forward to it!

**Comment by edwwaarrdd** [comment]
@andris-sevcenko 
I updated to 3.6.2 locally and tried the query again still null as a return when given a siteId other than 1.
I cleared all the log files and closed all tabs where the cms was active as to not add other stuff to the log.
And did a call from insomnia and the web.log file is 11461 lines long 😅.
What do you want me to look for? or can I email the file to you?

**Comment by andris-sevcenko** [comment]
@edwwaarrdd can you send a DB dump and your composer.json/lock files over to support@craftcms.com, while referencing this issue? I'll see if I can reproduce this locally and see what's what!

**Comment by edwwaarrdd** [comment]
@andris-sevcenko Sent the email with everything in it

**Comment by andris-sevcenko** [comment]
Thank you! Fixed for the next Craft 3.6 release!

**Comment by brandonkelly** [comment]
Craft 3.6.5 is out now with that fix.
